modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -254,7 +256,9 @@ public PollItemStatus accept(Consumer<PollItem<T, A>> consumer) {
       pollItem.validate();
 
       PollItemStatus status;
-      if (!acquireItem(pollItem, callbackContext)) {
+      if (isRequestedToStop()) {
```

Comment:
We should remove the other call to this method.

Reply:
nice catch. thanks.